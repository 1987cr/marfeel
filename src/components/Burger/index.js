import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.button`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 25px;
  height: 18px;
  background: transparent;
  border: none;
  cursor: pointer;

  &:focus {
    outline: none;
  }

  div {
    width: 25px;
    height: 3px;
    border-radius: 3px;
    background-color: ${({ color }) => color};
  }
`;

const Burger = ({ color, onClick }) => {
  return (
    <Container color={color} onClick={onClick}>
      <div />
      <div />
      <div />
    </Container>
  );
};

Burger.propTypes = {
  color: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Burger;
