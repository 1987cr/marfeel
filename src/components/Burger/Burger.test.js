import React from "react";
import { shallow } from "enzyme";
import Burger from "./";

describe("<Burger />", () => {
  const props = {
    color: "#fff",
    onClick: () => {}
  };

  it("renders correctly", () => {
    const wrapper = shallow(<Burger {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
