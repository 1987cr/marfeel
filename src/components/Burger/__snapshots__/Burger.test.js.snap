// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Burger /> renders correctly 1`] = `ShallowWrapper {}`;
